### PR TITLE
Image upload

### PR DESCRIPTION
**Pull request by peterpie123**
Lets users submit images along with their comments.

### CODE REVIEW

File: portfolio/src/main/java/com/google/sps/data/Comment.java [hunk 1]
```diff
@@ -44,27 +43,35 @@
   private final long id;
   /** The key that keeps this comment in persistent storage */
   private final Key key;
+  /** URL for an attached image. null if there is none */
+  private final String imageUrl;
 
   /** Create a comment with a posted date of given milliseconds from the epoch. */
-  public Comment(String text, String name, Key key, long timestamp) {
+  public Comment(String text, String name, Key key, long timestamp, String imageUrl) {
```
Comment (yutaow) [inline portfolio/src/main/java/com/google/sps/data/Comment.java:50]: Consider adding a @nullable annotation

File: portfolio/src/main/webapp/index.html [hunk 2]
```diff
@@ -52,43 +52,46 @@ <h2>A few facts about me</h2>
...
+      </form>
 		</div>
 
-		<hr />
+			<hr />
 
-		<div id="comment-header">
-			<h2 class="fa-stack">Comments</h2>
-			<button class="fas fa-trash fa-2x trash-inactive" id="comments-delete" onclick="deleteComments()"
+			<div id="comment-header">
```
Comment (yutaow) [inline portfolio/src/main/webapp/index.html:65]: Indentation looks strange in this file

File: portfolio/src/main/java/com/google/sps/data/CommentPersistHelper.java [hunk 3]
```diff
@@ -77,6 +89,44 @@ public void deleteComment(long id) {
...
+    ImagesService imagesService = ImagesServiceFactory.getImagesService();
+    ServingUrlOptions options = ServingUrlOptions.Builder.withBlobKey(blobKey);
+
+    // To support running in Google Cloud Shell with AppEngine's devserver, we must use the relative
+    // path to the image, rather than the path returned by imagesService which contains a host.
+    try {
+      URL url = new URL(imagesService.getServingUrl(options));
+      return url.getPath();
+    } catch (MalformedURLException e) {
+      return imagesService.getServingUrl(options);
```
Comment (davidgaskins) [inline portfolio/src/main/java/com/google/sps/data/CommentPersistHelper.java:126]: on a Malformed Error, we return the same thing? It seems like there could be a better way to handle this.

File: portfolio/src/main/java/com/google/sps/servlets/ImageServlet.java [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+public class ImageServlet extends HttpServlet {
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    BlobstoreService blobstoreService = BlobstoreServiceFactory.getBlobstoreService();
+    String uploadUrl = blobstoreService.createUploadUrl("/data");
+
+    response.setContentType("text/html");
+    response.getWriter().println(uploadUrl);
+  }
+}
```
Comment (davidgaskins) [inline portfolio/src/main/java/com/google/sps/servlets/ImageServlet.java:36]: newline here please :)

File: portfolio/src/main/java/com/google/sps/data/CommentPersistHelper.java (hunk 3 above)
Comment (peterpie123) [reply portfolio/src/main/java/com/google/sps/data/CommentPersistHelper.java:127]: A Malformed error is thrown when the path is relative (i.e. on a devserver), so the path is still valid, just not recognized as such by Java.

**Pull request merged**